confgenerator/confgenerator_test.go

```diff
@@ -129,39 +125,41 @@ func testGenerateConfsWithValidInput(t *testing.T, platform platformConfig) {
 			// Retrieve the expected golden conf files.
 			expectedMainConfig := readFileContent(t, platform.OS, testName, goldenMainPath, true)
 			expectedParserConfig := readFileContent(t, platform.OS, testName, goldenParserPath, true)
+
 			// Generate the actual conf files.
-			setFilepathJoin(platform.OS)
 			mainConf, parserConf, err := uc.GenerateFluentBitConfigs(platform.defaultLogsDir, platform.defaultStateDir, platform.InfoStat)
-			setFilepathJoin(runtime.GOOS)
 			if err != nil {
-				t.Fatalf("GenerateFluentBitConfigs got %v", err)
```

Comment:
In this function, it should remain `Fatalf`, because this test cannot proceed when the config generation fails. The test assertion is about the content of the generated config.